### PR TITLE
Completely remove vde_vmnet; socket_vmnet is the only supported option

### LINKED ISSUE #6248: Remove vde_vmnet support completely
In #6247 we made socket_vmnet the default. vde_vmnet is deprecated upstream and should be removed. It provides no functionality (that is used/exposed by Rancher Desktop) that isn't also available from socket_vmnet.

### PR DESCRIPTION
**Pull request by jandubois**
Closes #6248

### CODE REVIEW

**Comment by jandubois** [comment]
I ran a full BATS cycle with QEMU and containerd on macOS with the CI build, and it passed all 504 tests. I'm not running any other combinations right now, so don't wait for that 😄.

File: e2e/backend.e2e.spec.ts [hunk 1]
```diff
@@ -110,10 +110,7 @@ test.describe.serial('KubernetesBackend', () => {
         },
       };
       /** Platform-specific changes to `newSettings`. */
-      const platformSettings: Partial<Record<NodeJS.Platform, RecursivePartial<Settings>>> = {
-        win32:  { virtualMachine: { hostResolver: getAlternateSetting(currentSettings, 'virtualMachine.hostResolver', true, false) } },
-        darwin: { experimental: { virtualMachine: { socketVMNet: getAlternateSetting(currentSettings, 'experimental.virtualMachine.socketVMNet', true, false) } } },
```
Comment (mook-as) [inline e2e/backend.e2e.spec.ts:115]: This was here to test platform-specific settings; can we switch to a different platform-specific one instead, such as `useRosetta`?

File: e2e/backend.e2e.spec.ts [hunk 2]
```diff
@@ -143,10 +140,7 @@ test.describe.serial('KubernetesBackend', () => {
       };
 
       /** Platform-specific additions to `expectedDefinition`. */
-      const platformExpectedDefinitions: Partial<Record<NodeJS.Platform, ExpectedDefinition>> = {
-        win32:  { 'virtualMachine.hostResolver': false },
-        darwin: { 'experimental.virtualMachine.socketVMNet': false },
-      };
+      const platformExpectedDefinitions: Partial<Record<NodeJS.Platform, ExpectedDefinition>> = { win32: { 'virtualMachine.hostResolver': false } };
```
Comment (mook-as) [inline e2e/backend.e2e.spec.ts:143]: See above — we should still have a platform-specific thing (to test that things switch off correctly on Windows).

File: pkg/rancher-desktop/config/__tests__/settings.spec.ts [hunk 3]
```diff
@@ -553,9 +553,6 @@ describe('settings', () => {
             Debian:   false,
             openSUSE: true,
           },
-          experimental: {
```
Comment (mook-as) [inline pkg/rancher-desktop/config/__tests__/settings.spec.ts:556]: It doesn't make much sense to delete this: people with old config files would still have it, we need to test that we migrate it correctly (by dropping the field).

File: pkg/rancher-desktop/config/settingsImpl.ts [hunk 4]
```diff
@@ -399,7 +399,6 @@ export const updateTable: Record<number, (settings: any, locked : boolean) => vo
       { oldPath: 'telemetry', newPath: 'application.telemetry.enabled' },
       { oldPath: 'updater', newPath: 'application.updater.enabled' },
       { oldPath: 'kubernetes.containerEngine', newPath: 'containerEngine.name' },
-      { oldPath: 'kubernetes.experimental.socketVMNet', newPath: 'experimental.virtualMachine.socketVMNet' },
```
Comment (mook-as) [inline pkg/rancher-desktop/config/settingsImpl.ts:402]: This should stay (or be changed to a `_.unset()`).

File: pkg/rancher-desktop/config/settingsImpl.ts [hunk 5]
```diff
@@ -417,7 +416,6 @@ export const updateTable: Record<number, (settings: any, locked : boolean) => vo
       { oldPath: 'startInBackground', newPath: 'application.startInBackground' },
       { oldPath: 'window', newPath: 'application.window' },
       { oldPath: 'containerEngine.imageAllowList', newPath: 'containerEngine.allowedImages' },
-      { oldPath: 'virtualMachine.experimental.socketVMNet', newPath: 'experimental.virtualMachine.socketVMNet' },
```
Comment (mook-as) [inline pkg/rancher-desktop/config/settingsImpl.ts:420]: Same as above.

File: pkg/rancher-desktop/config/settingsImpl.ts
Comment (mook-as) [inline pkg/rancher-desktop/config/settingsImpl.ts:1]: There should be a new migration to delete the pref we're dropping.

File: pkg/rancher-desktop/config/settings.ts [hunk 6]
```diff
@@ -125,12 +125,10 @@ export const defaultSettings = {
     kubernetes:      { options: { spinkube: false } },
     virtualMachine:  {
       /** can only be set to VMType.VZ on macOS Ventura and later */
-      type:        VMType.QEMU,
+      type:       VMType.QEMU,
       /** can only be used when type is VMType.VZ, and only on aarch64 */
-      useRosetta:  false,
-      /** macOS only: if set, use socket_vmnet instead of vde_vmnet. */
-      socketVMNet: false,
```
Comment (mook-as) [inline pkg/rancher-desktop/config/settings.ts:132]: Should that lead to a settings version bump?

**Review by mook-as — CHANGES_REQUESTED** [review]

**Review by mook-as — APPROVED** [review]

**Pull request merged**